.github/workflows/test.yml

```diff
@@ -5,7 +5,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8, 3.9, '3.10', 3.11-dev, pypy2.7, pypy3.9]
+        python-version: ["3.5", "3.6", "3.7", "3.8", "3.9", "3.10", "3.11-dev", "pypy-3.7", "pypy-3.8", "pypy3.9"]
```

Comment:
This is a lot of CPU power.  It might be free in the sense of no cost but...  It will chew up electricity and heat the planet so let's be reasonable and not test all possible versions but only odd or even numbered versions.

Reply:
Tests are fast, so no reason to skip them.  @fangchenli but redundant quotes could be removed.  IIUIC, this is required only for "pypy-*" stuff.